### PR TITLE
add copyright header and job to check the headers

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
add copyright header and job to check the headers

#### Ticket Link
n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: hack/boilerplate/boilerplate.generatego.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Copyright (c) Mattermost, Inc. All Rights Reserved.
```
Comment (gigawhitlocks) [inline hack/boilerplate/boilerplate.generatego.txt:1]: Missing date range

File: hack/boilerplate/boilerplate.sh.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Copyright (c) YEAR-present Mattermost, Inc. All Rights Reserved.
```
Comment (gigawhitlocks) [inline hack/boilerplate/boilerplate.sh.txt:1]: YEAR-present

File: model/version.go [hunk 3]
```diff
@@ -1,5 +1,6 @@
-// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
```
Comment (gigawhitlocks) [inline model/version.go:1]: Did we intend to change this one from 2015 to 2020?

File: hack/boilerplate/boilerplate.generatego.txt (hunk 1 above)
Comment (cpanato) [reply hack/boilerplate/boilerplate.generatego.txt:1]: the pattern is left without a date when is generated files

File: hack/boilerplate/boilerplate.Makefile.txt [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Copyright (c) YEAR-present Mattermost, Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (cpanato) [reply hack/boilerplate/boilerplate.Makefile.txt:1]: this is the template to test against the files, should be like that

File: hack/boilerplate/boilerplate.sh.txt (hunk 2 above)
Comment (cpanato) [reply hack/boilerplate/boilerplate.sh.txt:1]: this is the template to test against the files, should be like that

File: model/version.go (hunk 3 above)
Comment (cpanato) [reply model/version.go:1]: just checked we need to use `2015-present` will update all

File: hack/boilerplate/boilerplate.generatego.txt (hunk 1 above)
Comment (cpanato) [reply hack/boilerplate/boilerplate.generatego.txt:1]: i'm going to change that to add that date as well

Comment (cpanato) [reply hack/boilerplate/boilerplate.generatego.txt]: done, thanks for your review @gigawhitlocks

File: hack/boilerplate/boilerplate.Makefile.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (gigawhitlocks) [reply hack/boilerplate/boilerplate.Makefile.txt:1]: my mistake!

**Pull request merged**